Realm.Shared/Realm.cs

```diff
@@ -676,6 +687,56 @@ public RealmResults<dynamic> All(string className)
             return new RealmResults<dynamic>(this, metadata, true);
         }
 
+
+        #region ById
+
+        private RealmObject.Metadata PrepById<T>(out IntPtr columnIndex)
```

Comment:
In the case where you'll need both the `columnIndex` and the `metadata`, I find the `out`-syntax cumbersome. Either return a tuple or simply return the metadata since it already contains the `ObjectIdColIndex` property and we can use that.